src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -124,6 +135,65 @@ export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFo
 
     const onCustomize = (e: MouseEvent) => {
       e.preventDefault();
+      if (isEmpty(template?.parameters)) {
+        return k8sCreate<V1Template>({
+          data: { ...template, metadata: { ...template?.metadata, namespace } },
+          model: ProcessedTemplatesModel,
+          ns: namespace,
+          queryParams: {
+            dryRun: 'All',
+          },
+        }).then(async (processedTemplate) => {
+          const vmObject = getTemplateVirtualMachineObject(processedTemplate);
+
+          const updatedVM = produce(vmObject, (vmDraft) => {
+            ensurePath(vmDraft, [
+              'spec.template.spec.domain.resources',
+              'spec.template.spec.domain.cpu',
+            ]);
+
+            vmDraft.metadata.namespace = namespace;
+            vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAME] = template.metadata.name;
+            vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAMESPACE] = template.metadata.namespace;
+            vmDraft.spec.template.spec.domain.resources.requests = {
+              ...vmDraft?.spec?.template?.spec?.domain?.resources?.requests,
+              memory: `${vm.spec.template.spec.domain.resources.requests['memory']}`,
+            };
+            vmDraft.spec.template.spec.domain.cpu.cores =
+              vmObject.spec.template.spec.domain.cpu.cores;
+
+            const updatedVolumes = applyCloudDriveCloudInitVolume(vmObject);
+            vmDraft.spec.template.spec.volumes = isRHELTemplate(processedTemplate)
+              ? updateCloudInitRHELSubscription(updatedVolumes, subscriptionData)
+              : updatedVolumes;
+          });
+
+          updateTabsData((tabsDataDraft) => {
+            // additional objects
+            tabsDataDraft.additionalObjects = processedTemplate.objects.filter((obj) =>
+              !isEmpty(authorizedSSHKey)
+                ? obj.kind !== VirtualMachineModel.kind || obj.kind !== SecretModel
+                : obj.kind !== VirtualMachineModel.kind,
+            );
+            // overview
+            ensurePath(tabsDataDraft, 'overview.templateMetadata');
+            tabsDataDraft.overview.templateMetadata.name = template.metadata.name;
+            tabsDataDraft.overview.templateMetadata.namespace = template.metadata.namespace;
+            tabsDataDraft.overview.templateMetadata.osType = getTemplateOS(template);
+            tabsDataDraft.overview.templateMetadata.displayName = getAnnotation(
+              template,
+              ANNOTATIONS.displayName,
+            );
+          });
+
+          // update context vm
+          await updateVM(
+            !isEmpty(authorizedSSHKey) ? addSecretToVM(updatedVM, authorizedSSHKey) : updatedVM,
+          );
+
+          history.push(`/k8s/ns/${namespace}/templatescatalog/review`);
+        });
+      }
```

Comment:
why we need it?
did i saw this function somewhere? :) 
why not just adding mandatory missing params?

Reply:
This is a case where the params doesn't exist at all in a template, so I want to prepare the vm for customization.
you probably saw something similar to this, but they do differ one each other.